_observing-your-data/prometheusmetrics.md

```diff
@@ -0,0 +1,116 @@
+
+![Prometheus example 2]({{site.url}}{{site.baseurl}}/images/metrics/metrics5.png)
+
+## Configuring a Prometheus data source connection to OpenSearch
+
+To configure a connection to Prometheus for metrics in OpenSearch Dashboards, create a file on your OpenSearch nodes named `datasources.json` containing the Prometheus data source settings. The following examples demonstrate the various Prometheus data source configurations using different authentication methods:
+
+No authentication:
+
+```json
+[{
+    "name" : "my_prometheus",
+    "connector": "prometheus",
+    "properties" : {
+        "prometheus.uri" : "http://localhost:9090"
+    }
+}]
+```
+
+Basic authentication:
+
+```json
+[{
+    "name" : "my_prometheus",
+    "connector": "prometheus",
+    "properties" : {
+        "prometheus.uri" : "http://localhost:9090",
+        "prometheus.auth.type" : "basicauth",
+        "prometheus.auth.username" : "admin",
+        "prometheus.auth.password" : "admin"
+    }
+}]
+```
+
+AWSSigV4 authentication:
+
+```json
+[{
+    "name" : "my_prometheus",
+    "connector": "prometheus",
+    "properties" : {
+        "prometheus.uri" : "http://localhost:8080",
+        "prometheus.auth.type" : "awssigv4",
+        "prometheus.auth.region" : "us-east-1",
+        "prometheus.auth.access_key" : "{{accessKey}}"
+        "prometheus.auth.secret_key" : "{{secretKey}}"
+    }
+}]
+```
+
+After configuring Prometheus in the `datasources.json` file, run the following command to load the configuration into the OpenSearch keystore. The configuration is securely stored as is contains credential information.
+
+```
+bin/opensearch-keystore add-file plugins.query.federation.datasources.config datasources.json
+```
+
+If you are updating the keystore during runtime, refresh the keystore using following API command:
+
+```bash
+POST /_nodes/reload_secure_settings
+{
+  "secure_settings_password":""
+  
+}
+```
+{% include copy-curl.html %}
+
+Metrics collected from Prometheus are displayed in the **Observability** application under **Metrics analytics**. For further information, see the [Prometheus Connector](https://github.com/opensearch-project/sql/blob/main/docs/user/ppl/admin/prometheus_connector.rst) GitHub page.
+
+## Prometheus PPL queries
+
+Below is an example of a [Piped Processing Language]({{site.url}}{{site.baseurl}}/search-plugins/sql/ppl/index) (PPL) query using a Prometheus connection in OpenSearch. This example demonstrates a metric selecting query with specific dimensions:
+
+```
+source = my_prometheus.prometheus_http_requests_total | stats avg(@value) by span(@timestamp,15s), handler, code
+```
+
+## Metrics Analytics UI
+
+The Metrics Analytics UI is a new component in the **Observability** application for OpenSearch Dashboards, as seen in the following image, that allows you to create visualizations based on metrics collected by your OpenSearch cluster.
+
+![Metrics UI example 1]({{site.url}}{{site.baseurl}}/images/metrics/metrics1.png)
```

Comment:
Saving visualizations? Although, saving is part of creating, so not sure if it would need to be called out here.

Reply:
It's more like, you can view a simple visualization, but you can also configure it with your own needs and then save the custom visualization. Let me see if I can explain that better.